Mathlib/Order/Directed.lean

```diff
@@ -250,18 +250,64 @@ theorem isBot_iff_isMin [IsDirected α (· ≥ ·)] : IsBot a ↔ IsMin a :=
 theorem isTop_iff_isMax [IsDirected α (· ≤ ·)] : IsTop a ↔ IsMax a :=
   ⟨IsTop.isMax, IsMax.isTop⟩
 
-variable (β) [PartialOrder β]
+end Preorder
+
+-- see Note [lower instance priority]
+instance (priority := 100) SemilatticeSup.to_isDirected_le [SemilatticeSup α] :
+    IsDirected α (· ≤ ·) :=
+  ⟨fun a b => ⟨a ⊔ b, le_sup_left, le_sup_right⟩⟩
+
+-- see Note [lower instance priority]
+instance (priority := 100) SemilatticeInf.to_isDirected_ge [SemilatticeInf α] :
+    IsDirected α (· ≥ ·) :=
+  ⟨fun a b => ⟨a ⊓ b, inf_le_left, inf_le_right⟩⟩
+
+-- see Note [lower instance priority]
+instance (priority := 100) OrderTop.to_isDirected_le [LE α] [OrderTop α] : IsDirected α (· ≤ ·) :=
+  ⟨fun _ _ => ⟨⊤, le_top _, le_top _⟩⟩
+
+-- see Note [lower instance priority]
+instance (priority := 100) OrderBot.to_isDirected_ge [LE α] [OrderBot α] : IsDirected α (· ≥ ·) :=
+  ⟨fun _ _ => ⟨⊥, bot_le _, bot_le _⟩⟩
+
+section PartialOrder
+
+variable [PartialOrder β]
 
-theorem exists_lt_of_directed_ge [IsDirected β (· ≥ ·)] [Nontrivial β] : ∃ a b : β, a < b := by
+section Nontrivial
+
+variable [Nontrivial β]
+
+theorem exists_lt_of_directed_ge (β) [PartialOrder β] [IsDirected β (· ≥ ·)] [Nontrivial β] :
```

Comment:
What changed here? Why do you have to repeat all variables?

Reply:
The variable `β` was accidentally being made explicit for all subsequent lemmas.

Reply:
You can use `variable (β) in` before each lemma where it should be explicit (or use `variable (β)`, then `variable {β}`). This way it's clear that you reuse TC assumptions from `variable`s, just change `β` to explicit.